rules-reviewed/eap7/eap6/hibernate4.windup.xml

```diff
@@ -604,5 +572,226 @@
                 <matches pattern="(TableGenerator|TableHiLoGenerator|SequenceGenerator|SequenceIdentityGenerator|SequenceHiLoGenerator)" />
             </where>
         </rule>
+        <rule id="hibernate4-00029">
+            <when>
+                <or>
+                    <javaclass references="org.hibernate.type.descriptor.sql.SqlTypeDescriptor">
+                        <location>IMPLEMENTS_TYPE</location>
+                    </javaclass>
+                    <javaclass references="org.hibernate.type.descriptor.java.JavaTypeDescriptor">
+                        <location>IMPLEMENTS_TYPE</location>
+                    </javaclass>
+                </or>
+                
+            </when>
+            <perform>
+                <hint title="Class org.hibernate.type.descriptor.sql.SqlTypeDescriptor change in Hibernate 5" effort="1" severity="mandatory">
+                    <message>Built-in `org.hibernate.type.descriptor.sql.SqlTypeDescriptor` or `org.hibernate.type.descriptor.java.JavaTypeDescriptor` implementations no longer auto-register themselves with `org.hibernate.type.descriptor.sql.SqlTypeDescriptorRegistry`.
```

Comment:
It's `SqlTypeDescriptorRegistry` for `SqlTypeDescriptor` and `JavaTypeDescriptorRegistry` for `JavaTypeDescriptor`.